linode/terraform-provider-linode
# new: Migrate `lke` data source to terraform framework

## 📝 Description

Migrate `lke` data source from SDKv2 to terraform framework. Also add some fields which are missed from the old schema. 

## ✔️ How to Test

`make testacc PKG_NAME="linode/lke"`

**Manual Test:** 
1. In a sandbox environment (i.e. dx-devenv), run the following terraform script to create a LKE cluster with 3 node pools:
```
resource "linode_lke_cluster" "test" {
    label       = "test-lke-cluster"
    region      = "br-gru"
    k8s_version = "1.28"
    tags        = ["test"]

    pool {
        type  = "g6-standard-2"
        count = 3
    }
}
```
2. Create a `linode_lke_cluster` data source. Assume that we wan to know the status of the above resource.
```
data "linode_lke_cluster" "test" {
    id = linode_lke_cluster.test.id
}

output "linode_lke_cluster" {
    value = data.linode_lke_cluster.test.status
}
```
3. Observe the data source can be created and read successfully. The status value is outputed.
